pkg/pubsub/system.go

```diff
@@ -0,0 +1,66 @@
+package pubsub
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/client"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/connection"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("pubsub")
+
+type System struct {
+	mux         sync.RWMutex
+	connections map[string]connection.Connection
+}
+
+var systems map[string]*System
```

Comment:
Do not use a global variable to cache these. Create a single system in main.go, then inject that system as needed:

* into the audit controller as you are currently doing with `auditDeps`
* into the controller described in `pubsub_init.go` (which should live under `pkg/controllers`), using controller.Dependencies (same model as `auditDeps`)

Again, there should be no need to use a global variable here, but if it is necessary for some reason I'm missing, it should be protected by a mutex.